perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/AppAutoRejectionScheduler.java

```diff
@@ -265,6 +265,14 @@ private void rejectApplication(Application application, String attrValueKey, Str
 
 		try {
 			registrarManager.rejectApplication(sess, application.getId(), message);
+			if (group == null && application.getUser() != null) {
```

Comment:
We should be able to handle situations where user object does not exist yet - we can identify the applications based on session info, which should then be filled in to fedInfo. i think we would ideally need to alter `filterPrincipalApplications` for that and call it with retrieved VO applications.